src/pdmat.jl

```diff
@@ -1,23 +1,49 @@
 """
 Full positive definite matrix together with a Cholesky factorization object.
 """
-struct PDMat{T<:Real,S<:AbstractMatrix} <: AbstractPDMat{T}
+struct PDMat{T<:Real,S<:AbstractMatrix{T}} <: AbstractPDMat{T}
     mat::S
     chol::Cholesky{T,S}
 
-    PDMat{T,S}(m::AbstractMatrix{T},c::Cholesky{T,S}) where {T,S} = new{T,S}(m,c)
+    function PDMat{T,S}(m::AbstractMatrix, c::Cholesky) where {T,S}
```

Comment:
Shouldn't this ensure
```suggestion
    function PDMat{T,S}(m::AbstractMatrix{T}, c::Cholesky{T,S}) where {T,S}
```
as on the master branch?

Reply:
No. This variant allows you to construct the chosen type regardless of the types of the inputs (as long as they are `AbstractMatrix` and `Cholesky`). Anything that doesn't match will be `convert`ed.